app/assets/javascripts/base/component.js

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable class-methods-use-this */
+
+//
```

Comment:
You can remove this line and the one in the end. So, all comments can be like this:

```javascript
// BaseComponent class to avoid boilerplate code
// into component classes and add some convention.
//
// The subclasses are not obligated to implement all
// of these methods but if they need to perform an action
// that fits one of the descriptions below, it's highly
// recommended to use the proper method.
```